design/construct-tree.md

```diff
@@ -0,0 +1,343 @@
+
+```bash
+├── App
+│   ├── Stack 1
+│   │   ├── Lambda Function [L2 Construct](... properties ...)
+|   │   │   ├── IAM Role (policy information)
+|   │   │   ├── Asset Code (Lambda Handler, pointer to S3 location where Asset will reside)
+|   │   │   ├── Lambda Function (Function definition and properties)
+```
+
+## Requirements
+
+This list describes only the minimal set of requirements from this feature. After we release these building blocks, we will look into extending the model to meet other use cases that arise.
+
+1. **Format** - Publish underlying data as a `.json` file that contains the information required to render a construct tree view. The construct tree illustrated in the previous section is only a depiction of what a human readable(ish) view might look like
+2. **Constructs**
+    1. ***Nodes*** - CDK application constructs at the root and drill down to the constructs it contains (1+ cloud resources.) Initially, we will focus our attention on AWS resources, but the model is extensible to cloud components from any provider
+    2. **Types/Hierarchy** - Types and hierarchy information will contain the details around the level of abstraction (L1, L2, L3..), service(s), and resource(s) that they represent.
+    3. ***Metadata -*** Constructs will expose metadata which will include an array of objects associated with the construct.
+        1. **Location** - The full path of the location where the construct is defined. This will support use cases where the construct-tree nodes can support navigability back to the declaration in the IDE itself.
+        2. ***Properties*** - Constructs will opt into the set of properties, and these will be exposed. Higher level constructs (L2 and L3) will expose properties and values of the resources they contain and not necessarily enumerate every property that’s available for a resource. This will especially be useful for constructs that are opinionated and set defaults on behalf of developers. Low level constructs (L1) will expose all properties that are configured.
+        3. ***Errors/Warnings*** - Errors and warnings that are produced during synthesis indicating validation failures, deprecation notices, guidance, etc will be included in the tree. These are emitted at the construct level although the message may point towards a specific property. The construct tree should be complete and include validation failures for all resources in the CDK application rather than just just fail on the first error.
+    4. **Assets** - Assets information will be included in the construct tree. Assets represent actions that the CDK *will* take ahead of stack deployments (i.e. S3 assets are zipped and uploaded directly, Docker images are uploaded to ECR). The asset metadata in the tree is the S3 key that an asset *would* have
```

Comment:
It's not clear to me from this section what information we'll be exporting to the JSON file. I understand there are 'assets' but what are the requirements for assets?